ignite/pkg/cosmosgen/generate.go

```diff
@@ -104,23 +117,141 @@ func (g *generator) setup() (err error) {
 			if err != nil {
 				return err
 			}
+			includes := []string{}
+			if len(modules) > 0 {
 
+				includes, err = g.resolveIncludes(path) // For versioning issues, we do dependency/includes resolution per module
+				if err != nil {
+					return err
+				}
+			}
 			modulesInPath = ModulesInPath{
-				Path:    path,
-				Modules: modules,
+				Path: path, // Each go module
+				ModuleIncludes: ModuleIncludes{ // Has a set of sdk modules and a set of includes to build them
+					Includes: includes,
+					Modules:  modules,
+				},
 			}
 
 			if err := moduleCache.Put(cacheKey, modulesInPath); err != nil {
 				return err
 			}
 		}
 
-		g.thirdModules[modulesInPath.Path] = append(g.thirdModules[modulesInPath.Path], modulesInPath.Modules...)
+		g.thirdModules[modulesInPath.Path] = modulesInPath.ModuleIncludes
 	}
 
 	return nil
 }
 
+func (g *generator) getProtoIncludeFolders(modPath string) []string {
+	// Add default protoDir and default includeDirs
+	includePaths := []string{filepath.Join(modPath, g.protoDir)}
+	for _, dir := range g.o.includeDirs {
+		includePaths = append(includePaths, filepath.Join(modPath, dir))
+	}
+	return includePaths
+}
+
+/*
+Leave this here for reference to improve resolution algorithm
+
+	func (g *generator) checkForProto(modpath string) (bool, error) {
+		err := filepath.WalkDir(modpath,
+			func(path string, _ fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				if filepath.Ext(path) == ".proto" {
+					return errFileFound
+				}
+				return nil
+			})
+		if err == errFileFound {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		return false, errFileNotFound
+	}
+*/
+func (g *generator) findBufPath(modpath string) (string, error) {
+	var bufPath string
+	err := filepath.WalkDir(modpath, func(path string, _ fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if filepath.Base(path) == "buf.yaml" {
+			bufPath = path
+			return filepath.SkipAll
+		}
+		return nil
+	})
+	if err != nil {
+		return "", err
+	}
+	return bufPath, nil
+}
+
+func (g *generator) generateBufIncludeFolder(modpath string) (string, error) {
+	protoPath, err := os.MkdirTemp("", "")
+	if err != nil {
+		return "", err
+	}
+
+	err = g.buf.Export(g.ctx, modpath, protoPath)
+	if err != nil {
+		os.Remove(protoPath)
```

Comment:
```suggestion
		os.RemoveAll(protoPath)
```

Reply:
No need to change it, I removed it in #3696